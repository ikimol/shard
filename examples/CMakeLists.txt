# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# shard :: examples :: CMakeLists.txt
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

# include CMake helper macros

include(${PROJECT_SOURCE_DIR}/cmake/add_target.cmake)

add_subdirectory(algorithm)
add_subdirectory(concurrency)
add_subdirectory(containers)
add_subdirectory(enums)
add_subdirectory(math)
add_subdirectory(memory)
add_subdirectory(property)
add_subdirectory(net)
add_subdirectory(random)
add_subdirectory(signal)
add_subdirectory(string)
add_subdirectory(system)
add_subdirectory(utility)
