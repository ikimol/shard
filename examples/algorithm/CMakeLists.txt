# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# shard :: examples :: algorithm :: CMakeLists.txt
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

shard_add_example(algorithm.enumerate
                  SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/enumerate.cpp
                  MODULES shard::algorithm
                  )
