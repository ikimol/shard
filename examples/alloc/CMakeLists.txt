# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# shard :: examples :: alloc :: CMakeLists.txt
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

shard_add_example(alloc.allocators
                  SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/allocators.cpp
                  MODULES shard::alloc
                  )

shard_add_example(alloc.containers.array
                  SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/array.cpp
                  MODULES shard::alloc
                  )

shard_add_example(alloc.containers.stack
                  SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/stack.cpp
                  MODULES shard::alloc
                  )
