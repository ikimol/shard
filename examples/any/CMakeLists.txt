# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# shard :: examples :: any :: CMakeLists.txt
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

shard_add_example(any
                  SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/any.cpp
                  MODULES shard::any
                  )
