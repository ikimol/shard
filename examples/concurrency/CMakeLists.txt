# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# shard :: examples :: concurrency :: CMakeLists.txt
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

shard_add_example(concurrency.channel
                  SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/channel.cpp
                  MODULES shard::concurrency
                  )

shard_add_example(concurrency.sync
                  SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/thread_safe.cpp
                  MODULES shard::concurrency
                  )
