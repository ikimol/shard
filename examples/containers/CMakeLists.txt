# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# shard :: examples :: containers :: CMakeLists.txt
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

shard_add_example(containers.sparse_set
                  SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/sparse_set.cpp
                  MODULES shard::containers shard::meta
                  )
