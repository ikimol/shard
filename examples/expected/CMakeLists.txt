# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# shard :: examples :: expected :: CMakeLists.txt
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

shard_add_example(expected
                  SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/expected.cpp
                  MODULES shard::expected
                  )